mds/entrypoint.sh

```diff
@@ -34,6 +47,27 @@ if [ ! -e /var/lib/ceph/mds/ceph-$MDS_NAME/keyring ]; then
       # Generate the new MDS key
       ceph auth get-or-create mds.$MDS_NAME mds 'allow' osd 'allow *' mon 'allow profile mds' > /var/lib/ceph/mds/ceph-${MDS_NAME}/keyring
    fi
+
+fi
+
+# Create the Ceph filesystem, if necessary
+if [ $CEPHFS_CREATE -gt 0 ]; then
+   FS_EXISTS=$(ceph fs ls | grep -c name:.${CEPHFS_NAME},)
+   if [ $FS_EXISTS -eq 0 ]; then
+      # Make sure the specified data pool exists
+      ceph osd pool stats ${CEPHFS_DATA_POOL} >/dev/null
+      if [ $? -ne 0 ]; then
+         ceph osd pool create ${CEPHFS_DATA_POOL} ${CEPHFS_DATA_POOL_PG}
+      fi
+
+      # Make sure the specified metadata pool exists
```

Comment:
So, ultimately:
1) don't do _anything_ unless we are told to create the filesystem
2) don't do anything if the filesystem already exists
3) Otherwise, create each pool if _they_ don't exist
4) Create the filesystem